src/enzyme_ad/jax/Dialect/Ops.cpp

```diff
@@ -57,12 +57,46 @@ KernelCallOp::verifySymbolUses(SymbolTableCollection &symbolTable) {
   return success();
 }
 
+void KernelCallOp::setCalleeFromCallable(mlir::CallInterfaceCallable callee) {
+  auto symbol = cast<SymbolRefAttr>(callee);
+  setFnAttr(cast<FlatSymbolRefAttr>(symbol));
+}
+
 CallInterfaceCallable KernelCallOp::getCallableForCallee() {
   return SymbolRefAttr::get(getContext(), getFn());
 }
 
 Operation::operand_range KernelCallOp::getArgOperands() { return getInputs(); }
 
+mlir::MutableOperandRange KernelCallOp::getArgOperandsMutable() {
+  MutableOperandRange range(*this);
+  return range;
+}
+
+mlir::ArrayAttr KernelCallOp::getArgAttrsAttr() {
+  mlir::ArrayAttr arr;
```

Comment:
I would simply return nullptr. Same for the other methods where you use this pattern.